src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -223,6 +223,14 @@ void Cancel()
                     if (ipAddresses[i] != null)
                     {
                         sockets[i] = new Socket(ipAddresses[i].AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+
+                        // enable keep-alive on socket
+                        const int time = 30;
```

Comment:
This is the code block I added to enable keep alive on the SqlClient sockets.  The rest of the code is a backport of the framework keep alive support from master into the release/2.1 branch which is required in order to set the option.